library/ReplicatedTableConfigurationManager.cs

```
@@ -60,33 +64,67 @@ private void NewStrategyToSetConnectionString(ReplicaInfo replica)
             {
                 replica.ConnectionString = null;
             }
+
+            return GetConnectionStringTableClientForReplica(replica);
         }
 
         /// <summary>
-        /// Connection string infered from blob content
+        /// Connection string inferred from blob content
         /// </summary>
         /// <param name="replica"></param>
-        private void OldStrategyToSetConnectionString(ReplicaInfo replica)
+        private TableServiceClient OldTableClientFromConnectionStringStrategy(ReplicaInfo replica)
         {
             string connectionString = String.Format(Constants.ShortConnectioStringTemplate, useHttps ? "https" : "http", replica.StorageAccountName, replica.StorageAccountKey);
             replica.ConnectionString = SecureStringHelper.ToSecureString(connectionString);
+
+            return GetConnectionStringTableClientForReplica(replica);
         }
 
+        /// <summary>
+        /// Managed Identity provided by user.
+        /// </summary>
+        /// <param name="replica"></param>
+        /// <returns></returns>
+        private TableServiceClient TableClientFromManagedIdentityStrategy(ReplicaInfo replica)
+        {
+            TokenCredential storageToken = null;
+            if (this.replicaStorageTokenMap.ContainsKey(replica.StorageAccountName))
+            {
+                storageToken = this.replicaStorageTokenMap[replica.StorageAccountName];
+            }
```

Although this case shouldn't occur where replica storage account name isn't found in replicaStorageTokenMap, should we throw an exception in this case to make debugging easier if there happens to be some gap?

---

In View.cs, where these strategies are used, there's logging to print replica info on error.

We should actually avoid throwing here in favor of returning a null `TableServiceClient`. We want to return a null client so the error is logged and the view chain is cleared.
https://github.com/thbui-msft/rtable/blob/45adb850cbe9ad7c712df547cd089a750873dedb/library/View.cs#L107-L114